File: tests/fixtures/java/UserService.java
package com.example.service;

import java.util.Optional;
import java.util.UUID;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class UserService {

    private final UserRepository repository;

    public UserService(UserRepository repository) {
        this.repository = repository;
    }

    public Optional<User> findById(UUID id) {
        return repository.findByIdAndIsDeletedIsFalse(id);
    }

    @Transactional
    public User create(User user, String createdBy) {
        if (user.getId() != null) {
            throw new UnsupportedOperationException("ID must be null for creation");
        }
        validateUser(user);
        return repository.save(user);
    }

    @Transactional
    public User update(User user, String modifiedBy) {
        if (user.getId() == null) {
            throw new UnsupportedOperationException("ID must not be null for update");
        }
        return repository.save(user);
    }

    private void validateUser(User user) {
        if (user.getName() == null || user.getName().isBlank()) {
            throw new IllegalArgumentException("Name is required");
        }
    }
}

File: tests/fixtures/kotlin/UserService.kt
package com.example.svc

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

interface UserRepository {
    fun findById(id: String): User?
    fun save(user: User): User
}

data class User(val id: String, val name: String)

@Service
class UserService(private val repo: UserRepository) {

    @Transactional
    fun create(name: String): User {
        val u = User(id = "x", name = name)
        return repo.save(u)
    }

    fun find(id: String): User? = repo.findById(id)
}

object Constants {
    const val DEFAULT_PAGE_SIZE = 20
}

fun topLevelHelper(a: Int, b: Int): Int = a + b
